configure() in libqtile.backend.x11.xcbq is not robust

<!--
Please do not ask general questions here! There are [community
contact](https://github.com/qtile/qtile#community) options for that.
-->

# Issue description

The function is not robust, as it breaks on xcffib declaring a version with patch level:
https://github.com/qtile/qtile/blob/464bf7f191e800bcb8a075d465841cde81bc15f8/libqtile/backend/x11/xcbq.py#L619-L628

# Qtile version

0.16.1

# Stack traces

Testing breaks on this:

```
>           raise AssertionError("Error launching Qtile, traceback:\n%s" % error)
E           AssertionError: Error launching Qtile, traceback:
E           Traceback (most recent call last):
E             File "/build/qtile/src/qtile-0.16.1/test/conftest.py", line 286, in run_qtile
E               q = SessionManager(kore, config_class(), fname=self.sockfile)
E             File "/build/qtile/src/qtile-0.16.1/libqtile/core/session_manager.py", line 31, in __init__
E               self.qtile = Qtile(kore, config, self.eventloop, no_spawn=no_spawn, state=state)
E             File "/build/qtile/src/qtile-0.16.1/libqtile/core/manager.py", line 161, in __init__
E               self._process_screens()
E             File "/build/qtile/src/qtile-0.16.1/libqtile/core/manager.py", line 352, in _process_screens
E               scr._configure(
E             File "/build/qtile/src/qtile-0.16.1/libqtile/config.py", line 289, in _configure
E               i._configure(qtile, self)
E             File "/build/qtile/src/qtile-0.16.1/libqtile/bar.py", line 208, in _configure
E               self.window = window.Internal.create(
E             File "/build/qtile/src/qtile-0.16.1/libqtile/window.py", line 688, in create
E               i.place(x, y, width, height, 0, None)
E             File "/build/qtile/src/qtile-0.16.1/libqtile/window.py", line 492, in place
E               self.window.configure(**kwarg)
E             File "/build/qtile/src/qtile-0.16.1/libqtile/backend/x11/xcbq.py", line 623, in configure
E               if float(xcffib.__xcb_proto_version__) < 1.12:
E           ValueError: could not convert string to float: '1.14.1'
```
# Configuration

n/a

--- a/libqtile/backend/x11/xcbq.py
+++ b/libqtile/backend/x11/xcbq.py
@@ -623,7 +623,7 @@ class Window:
         mask, values = ConfigureMasks(**kwargs)
         # older versions of xcb pack everything into unsigned ints "=I"
         # since 1.12, uses switches to pack things sensibly
-        if float(xcffib.__xcb_proto_version__) < 1.12:
+        if float(".".join(xcffib.__xcb_proto_version__.split(".")[0: 2])) < 1.12:
             values = [i & 0xffffffff for i in values]
         return self.conn.conn.core.ConfigureWindow(self.wid, mask, values)
 

